modules/pricer/src/main/java/com/opengamma/strata/pricer/impl/tree/OptionFunction.java

```diff
@@ -0,0 +1,148 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ * 
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.pricer.impl.tree;
+
+import java.util.Arrays;
+
+import com.opengamma.strata.collect.array.DoubleArray;
+import com.opengamma.strata.collect.array.DoubleMatrix;
+
+/**
+ * Option function interface used in tree option pricing.
+ */
+public interface OptionFunction {
+
+  /**
+   * Obtains strike value.
+   * 
+   * @return strike
+   */
+  public abstract double getStrike();
```

Comment:
Is getStrike part of the 'OptionFunction' interface or should it be at the specific implementation level, e.g. in 'EuropeanVanillaOptionFunction'? We could imagine pricing option that don't have a single strike, like AD options or straddle, condor, etc.